BUILD

```diff
@@ -6700,6 +6699,7 @@ grpc_cc_library(
     tags = ["nofixdeps"],
     visibility = ["@grpc:public"],
     deps = [
+        "grpc++_base",
```

Comment:
I think this is a better place to bring the dependency given that we are removing `grpc++_codegen_base` and `grpc++_codegen_base_src` in the future.

Also, this change makes the blaze rule in Google internal not need to add the dependency.